pkg/image/api/helper.go

```diff
@@ -313,12 +314,24 @@ func ManifestMatchesImage(image *Image, newManifest []byte) (bool, error) {
 	if err != nil {
 		return false, err
 	}
-	sm := schema1.SignedManifest{Raw: newManifest}
-	raw, err := sm.Payload()
-	if err != nil {
-		return false, err
+	var canonical []byte
+	if image.DockerImageManifestMediaType == schema2.MediaTypeManifest {
+		var m schema2.DeserializedManifest
+		if err := json.Unmarshal(newManifest, &m); err != nil {
+			return false, err
+		}
+		_, canonical, err = m.Payload()
+		if err != nil {
+			return false, err
+		}
+	} else {
```

Comment:
I would match the schema1 media type specifically and handle other other media types in yet another branch.